doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,110 @@
+$ApiDependenciesDir = "$($FrameworkDir)/dependencies/my-api"
+$XmlOutDir = "$($BinDirectory)/$($ArtifactName)/dll-xml-output"
+$YamlOutDir = "$($BinDirectory)/$($ArtifactName)/dll-yaml-output"
+$DocOutDir = "$($BinDirectory)/$($ArtifactName)/docfx-output/docfx_project"
+
+if ($LibType -eq '') { 
+    Write-Verbose "Build Packages for Doc Generation - Client"
+    dotnet build "$($RepoRoot)/eng/service.proj" /p:ServiceDirectory=$PackageLocation /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath=$ApiDir /p:TargetFramework=netstandard2.0
+
+    Write-Verbose "Include client Dependencies"
+    dotnet build "$($RepoRoot)/eng/service.proj" /p:ServiceDirectory=$PackageLocation /p:IncludeTests=false /p:IncludeSamples=false /p:OutputPath=$ApiDependenciesDir /p:TargetFramework=netstandard2.0 /p:CopyLocalLockFileAssemblies=true
+}
+
+if ($LibType -eq 'Management') { # Management Package
+    Write-Verbose "Build Packages for Doc Generation - Management"
+    dotnet msbuild "eng/mgmt.proj" /p:scope=$PackageLocation /p:OutputPath=$ApiDir -maxcpucount:1 -nodeReuse:false
+
+    Write-Verbose "Include Management Dependencies"
+    dotnet msbuild "eng/mgmt.proj" /p:scope=$PackageLocation /p:OutputPath=$ApiDependenciesDir /p:CopyLocalLockFileAssemblies=true -maxcpucount:1 -nodeReuse:false
+}
+
+Write-Verbose "Remove all unneeded artifacts from build output directory"
+Remove-Item –Path "$($ApiDir)/*" -Include * -Exclude "$($ArtifactName).dll", "$($ArtifactName).xml"
+
+Write-Verbose "Initialize Frameworks File"
+& "$($BinDirectory)/mdoc/mdoc.exe" fx-bootstrap "$($FrameworkDir)"
+
+Write-Verbose "Include XML Files"
+& "$($BinDirectory)/PopImport/popimport.exe" -f "$($FrameworkDir)"
+
+Write-Verbose "Produce ECMAXML"
+& "$($BinDirectory)/mdoc/mdoc.exe" update -fx "$($FrameworkDir)" -o "$($XmlOutDir)" --debug -lang docid -lang vb.net -lang fsharp --delete
+
+Write-Verbose "Generate YAML"
+& "$($BinDirectory)/ECMA2Yml/ECMA2Yaml.exe" -s "$($XmlOutDir)" -o "$($YamlOutDir)"
+
+Write-Verbose "Provision DocFX Directory"
+& "$($BinDirectory)/docfx/docfx.exe" init -q -o "$($DocOutDir)"
+
+Write-Verbose "Copy over Package ReadMe"
+$PkgReadMePath = "$($RepoRoot)/sdk/$($PackageLocation)/README.md"
+if ([System.IO.File]::Exists($PkgReadMePath))
+{
+    Copy-Item $PkgReadMePath -Destination "$($DocOutDir)/api/index.md" -Force
+    Copy-Item $PkgReadMePath -Destination "$($DocOutDir)/index.md" -Force
+}
+else
+{
+    New-Item "$($DocOutDir)/api/index.md" -Force
+    Add-Content -Path "$($DocOutDir)/api/index.md" -Value "This Package Contains no Readme."
+    Copy-Item "$($DocOutDir)/api/index.md" -Destination "$($DocOutDir)/index.md"-Force
+    Write-Verbose "Package ReadMe was not found"
+}
+
+Write-Verbose "Copy over generated yml and other assets"
+Copy-Item "$($YamlOutDir)/*"-Destination "$($DocOutDir)/api" -Recurse
+Copy-Item "$($RepoRoot)/doc/ApiDocGeneration/assets/docfx.json" -Destination "$($DocOutDir)" -Recurse -Force
```

Comment:
Capture `$($RepoRoot)/doc/ApiDocGeneration` in a variable. In general whenever you see duplication it is a good places to consider capturing a variable.